lib/ex_unit/lib/ex_unit/case.ex

```diff
@@ -314,7 +321,18 @@ defmodule ExUnit.Case do
       Module.put_attribute(mod, :ex_unit_test_names, Map.put(test_names, name, true))
     end
 
-    Module.delete_attribute(mod, :tag)
+    Enum.each [:tag | test_attributes], fn(attribute) ->
+      Module.delete_attribute(mod, attribute)
+    end
+  end
+
+  @doc """
+  """
+  defmacro register_attribute(mod, new, opts \\ []) when is_atom(new) do
+    quote do
+      Module.register_attribute(unquote(mod), unquote(new), unquote(opts))
+      @test_attributes unquote(new)
```

Comment:
I think we should prefix it with `ex_unit` (like we do for other attributes) to avoid possible clashed with used-defined.